src/utils/getRowsInternal.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  GetRowsCallback,
+  GetRowsOptions,
+  GetRowsResponse,
+  TabularApiSurface,
+} from '../tabular-api-surface';
+import {createReadStreamInternal} from './createReadStreamInternal';
+import {Row} from '../row';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const concat = require('concat-stream');
+
+/**
+ * Get {@link Row} objects for the rows currently in your table.
+ *
+ * This method is not recommended for large datasets as it will buffer all rows
+ * before returning the results. Instead we recommend using the streaming API
+ * via {@link Table#createReadStream}.
+ *
+ * @param {TabularApiSurface} table The table instance to get rows from.
+ * @param {boolean} singleRow Boolean to check if the request is for a single row.
+ * @param {string} [viewName] The name of the authorized view, if applicable.
+ * @param {object} [optionsOrCallback] Configuration object. See
+ *     {@link Table#createReadStream} for a complete list of options.
+ * @param {object} [optionsOrCallback.gaxOptions] Request configuration options, outlined
+ *     here: https://googleapis.github.io/gax-nodejs/CallSettings.html.
+ * @param {function} cb The callback function.
+ * @param {?error} cb.err An error returned while making this request.
+ * @param {Row[]} cb.rows List of Row objects.
+ *
+ * @returns {Promise<GetRowsResponse>|void} Returns a promise that resolves with the rows if no callback is provided, otherwise calls the callback with the rows.
+ *
+ * @example <caption>include:samples/api-reference-doc-snippets/table.js</caption>
+ * region_tag:bigtable_api_get_rows
+ */
+export function getRowsInternal(
```

Comment:
why do we need getRowsInternal? Can we call createReadStreamInternal directly?

Reply:
row.get used to call getRows. Since getRows calls createReadStream for both single rows and multi row calls then getRows needs a `singleRow` parameter so should be pulled out into getRowsInternal.

In `row.get` we should not call `createReadStreamInternal` because then we need to rewrite the `getRows` function inside `row.get` so that `row.get` does what it did before. This will make it so that we have to implement the `getRows` logic in two places which isn't ideal.